integration/container_interface.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils/testutils"
+)
+
+type ContainerSpec struct {
+	Name         string
+	Cmd          string
+	Options      []containerOption
+	Cleanup      bool
+	StartAndStop bool
+}
+
+type ContainerFactory interface {
+	NewContainer(ContainerSpec) ContainerInterface
+}
+
+type ContainerInterface interface {
+	Run(t *testing.T)
+	Start(t *testing.T)
+	Stop(t *testing.T)
+	IsCleanup() bool
+	IsStartAndStop() bool
+	Running() bool
+}
+
+func NewContainerFactory(runtime string) ContainerFactory {
+	switch runtime {
+	case "docker":
+		return &DockerManager{}
+	case "containerd":
+		return &ContainerdManager{}
+	default:
+		panic("Unknown runtime " + runtime)
```

Comment:
Since this is a test util do you think it makes sense to use [runtime.Goexit](https://pkg.go.dev/runtime#Goexit) here ?

Reply:
I would like to have a message somewhere that the runtime given is unknown. `Goexit` doesn't take a string. I also looked into having a `testing.T.Errorf` but the time where we call `NewContainerFactory` doesn't allow that? We only have a `testing.M` instance there.
Maybe just `fmt.Println`?